src/interpreter/initialization.rs

```diff
@@ -53,6 +50,15 @@ impl<S> Interpreter<S> {
 
         Ok(())
     }
+
+    /// Initialize the VM for a predicate context
+    pub fn init_predicate(&mut self, tx: CheckedTransaction) -> bool {
+        self.context = Context::Predicate {
+            program: Default::default(),
```

Comment:
why use a default value here instead of just passing through the idx of the predicate?